tests/components/kraken/test_sensor.py

```diff
@@ -255,13 +260,25 @@ async def test_missing_pair_marks_sensor_unavailable(hass):
             assert sensor.state == "0.0003494"
 
         with patch(
-            "pykrakenapi.KrakenAPI.get_ticker_information",
-            side_effect=KrakenAPIError("EQuery:Unknown asset pair"),
+            "pykrakenapi.KrakenAPI.get_tradable_asset_pairs",
+            return_value=MISSING_PAIR_TRADEABLE_ASSET_PAIR_RESPONSE,
         ):
-            async_fire_time_changed(
-                hass, utcnow + timedelta(seconds=DEFAULT_SCAN_INTERVAL * 2)
-            )
-            await hass.async_block_till_done()
-
-            sensor = hass.states.get("sensor.xbt_usd_ask")
-            assert sensor.state == "unavailable"
+            with patch(
+                "pykrakenapi.KrakenAPI.get_ticker_information",
+                side_effect=KrakenAPIError("EQuery:Unknown asset pair"),
+            ):
+                async_fire_time_changed(
+                    hass, utcnow + timedelta(seconds=DEFAULT_SCAN_INTERVAL * 2)
+                )
+                await hass.async_block_till_done()
+                with patch(
```

Comment:
This looks to be re-patching same api in a nested form.

 Just update the mock object instead.

with patch() as mocked:
  do stuff...

  mock.side_effects = None
  mock.return_value = xx

Reply:
Thanks! Done